purchase_order_approved/models/purchase_order.py

```diff
@@ -28,14 +28,16 @@ class PurchaseOrder(models.Model):
     company_id = fields.Many2one(states=READONLY_STATES)
     picking_type_id = fields.Many2one(states=READONLY_STATES)
 
-    @api.multi
     def button_release(self):
-        super(PurchaseOrder, self).button_approve()
+        for rec in self:
+            super(PurchaseOrder, rec).button_approve()
 
-    @api.multi
     def button_approve(self, force=False):
-        approve_purchases = self.filtered(
-            lambda p: p.company_id.purchase_approve_active)
-        approve_purchases.write({'state': 'approved'})
-        return super(PurchaseOrder, self - approve_purchases).button_approve(
-            force=force)
+        for rec in self:
```

Comment:
It should have been modified. It is not necessary to go for records as you can write on the hole recordset at once and then run button approve function that also allows multi write.

Reply:
I see, Thank for reviews

Reply:
@HviorForgeFlow I see, Thank for reviews